### PR TITLE
Refactor DocumentManager and LSCompiler API

### PR DESCRIPTION
**Pull request by rasika**

### CODE REVIEW

File: composer/modules/server/services/ballerina-parser/src/main/java/org/ballerinalang/composer/service/ballerina/parser/service/JSONModelConstants.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*
+*  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (nadeeshaan) [inline composer/modules/server/services/ballerina-parser/src/main/java/org/ballerinalang/composer/service/ballerina/parser/service/JSONModelConstants.java]: Invalid License Header

File: language-server/modules/langserver-compiler/src/test/java/org/ballerinalang/langserver/compiler/workspace/WorkspaceDocumentManagerImplTest.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
+package org.ballerinalang.langserver.compiler.workspace;
```
Comment (nadeeshaan) [inline language-server/modules/langserver-compiler/src/test/java/org/ballerinalang/langserver/compiler/workspace/WorkspaceDocumentManagerImplTest.java]: Missing License header

File: composer/modules/server/services/ballerina-parser/src/main/java/org/ballerinalang/composer/service/ballerina/parser/service/JSONModelConstants.java (hunk 1 above)
Comment (rasika) [reply composer/modules/server/services/ballerina-parser/src/main/java/org/ballerinalang/composer/service/ballerina/parser/service/JSONModelConstants.java]: It is a rename of the file "BLangJSONModelConstants.java" into "JSONModelConstants.java" along with some edits. I suppose we can keep the license header as it is.

File: language-server/modules/langserver-compiler/src/test/java/org/ballerinalang/langserver/compiler/workspace/WorkspaceDocumentManagerImplTest.java (hunk 2 above)
Comment (rasika) [reply language-server/modules/langserver-compiler/src/test/java/org/ballerinalang/langserver/compiler/workspace/WorkspaceDocumentManagerImplTest.java]: Will fix it

File: composer/modules/server/services/ballerina-parser/src/main/java/org/ballerinalang/composer/service/ballerina/parser/service/util/BLangFragmentParser.java [hunk 3]
```diff
@@ -137,15 +137,16 @@ protected static JsonObject getJsonNodeForFragment(JsonObject jsonModel, BLangSo
...
-        BLangCompilationUnit compilationUnit = null;
-        java.nio.file.Path filePath = LSCompiler.createAndGetTempFile(UNTITLED_BAL);
-        BallerinaFile model = LSCompiler.compileContent(source, filePath, CompilerPhase.DEFINE, documentManager, true);
-        if (model.getBLangPackage() != null) {
-            compilationUnit = model.getBLangPackage().getCompilationUnits().stream().
-                    filter(compUnit -> UNTITLED_BAL.equals(compUnit.getName())).findFirst().get();
-        }
-        return TextDocumentFormatUtil.generateJSON(compilationUnit, new HashMap<>());
+            throws LSCompilerException, JSONGenerationException {
+        java.nio.file.Path filePath = LSCompilerUtil.createTempFile(LSCompilerUtil.UNTITLED_BAL);
```
Comment (nadeeshaan) [inline composer/modules/server/services/ballerina-parser/src/main/java/org/ballerinalang/composer/service/ballerina/parser/service/util/BLangFragmentParser.java]: Any specific reason to add the full package path here?

Comment (rasika) [reply composer/modules/server/services/ballerina-parser/src/main/java/org/ballerinalang/composer/service/ballerina/parser/service/util/BLangFragmentParser.java]: Nope, fixed with https://github.com/ballerina-platform/ballerina-lang/pull/9836/commits/b578fda8ac13fe77028c276521a36f6be7a2067b

File: language-server/modules/langserver-compiler/src/test/java/org/ballerinalang/langserver/compiler/workspace/WorkspaceDocumentManagerImplTest.java (hunk 2 above)
Comment (rasika) [reply language-server/modules/langserver-compiler/src/test/java/org/ballerinalang/langserver/compiler/workspace/WorkspaceDocumentManagerImplTest.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/9836/commits/b6cca5194c5bcef685f1c8104c5ab992ee8f9dfa

**Pull request merged**